File: vuln-sandbox/node/cwe-347/jwt-none-algorithm-safe.js
// SAFE: jwt-none-algorithm — algorithm explicitly set to HS256; 'none' not accepted
// Rule: CryptoJwtNoneAlgorithm | CWE-347 | Expected: TrueNegative

const express = require('express');
const jwt = require('jsonwebtoken');

const app = express();

app.get('/verify', (req, res) => {
  const token = req.headers.authorization?.split(' ')[1];
  if (!token) return res.status(401).json({ error: 'No token' });

  // SAFE: algorithms array explicitly excludes 'none'; only HS256 is accepted
  jwt.verify(token, process.env.JWT_SECRET, { algorithms: ['HS256'] }, (err, decoded) => {
    if (err) return res.status(401).json({ error: 'Invalid token' });
    res.json({ user: decoded });
  });
});

app.listen(3000);

File: vuln-sandbox/python/cwe-347/jwt-none-algorithm-safe.py
# SAFE: jwt-none-algorithm — algorithm explicitly set to HS256; 'none' not accepted
# Rule: CryptoJwtNoneAlgorithm | CWE-347 | Expected: TrueNegative

import os
import jwt
from flask import Flask, request, jsonify

app = Flask(__name__)
SECRET = os.environ['JWT_SECRET']


@app.route('/verify')
def verify():
    token = request.headers.get('Authorization', '').removeprefix('Bearer ')
    if not token:
        return jsonify({'error': 'No token'}), 401

    # SAFE: algorithms list explicitly restricts to HS256; 'none' is not accepted
    try:
        payload = jwt.decode(token, SECRET, algorithms=['HS256'])
    except jwt.InvalidTokenError as e:
        return jsonify({'error': str(e)}), 401

    return jsonify({'user': payload})


if __name__ == '__main__':
    app.run()
